clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -2538,6 +2538,35 @@ static mlir::LLVM::CallIntrinsicOp replaceOpWithCallLLVMIntrinsicOp(
   return callIntrinOp;
 }
 
+class CIRIntrinsicCallLowering
+    : public mlir::OpConversionPattern<mlir::cir::IntrinsicCallOp> {
+public:
+  using OpConversionPattern<mlir::cir::IntrinsicCallOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::IntrinsicCallOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    mlir::Type llvmResTy =
+        getTypeConverter()->convertType(op->getResultTypes()[0]);
+    if (!llvmResTy)
+      return op.emitError("expected LLVM result type");
+    StringRef name = op.getIntrinsicName();
+    // Some llvm intrinsics require ElementType attribute to be attached to
+    // the argument of pointer type. That prevents us from generating LLVM IR
+    // because from LLVM dialect, we have LLVM IR like the below which fails
+    // LLVM IR verification.
+    // %3 = call i64 @llvm.aarch64.ldxr.p0(ptr %2)
+    // The expected LLVM IR should be like
+    // %3 = call i64 @llvm.aarch64.ldxr.p0(ptr elementtype(i32) %2)
+    // MLIR LLVM dialect should handle this part as CIR has no way
```

Comment:
`// MLIR LLVM dialect` -> use a `// TODO(cir): MLIR LLVM dialect` or `FIXME` here, since this is something we want to handle in the future.